### PR TITLE
Add code and tests for Task 10

### PR DESCRIPTION
**Pull request by solveigag**
Finally solved the promises mystery! :partying_face:

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -375,4 +376,68 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+          expect(body.msg).toBe("Missing required fields.");
+        });
+    });
+  });
 })
```
Comment (corkesw) [inline __tests__/app.test.js:453]: Good tests... what if the user doesn't exisit?

File: app.js [hunk 2]
```diff
@@ -16,18 +17,21 @@ app.get("/api/articles/:article_id", getArticleById);
...
 app.use((err, req, res, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02" || err.code === "23503") {
     res.status(400).send({ msg: "Invalid input data type" });
   } else if (err.code === "23502") {
     res.status(400).send({ msg: "Missing required fields." });
-  } else if (err.status) {
+  // } else if (err.code === "23503") {
+  //   res.status(404).send({msg: "Article not found!"});
+} else if (err.status) {
```
Comment (corkesw) [inline app.js:32]: Why is this code commented out?

Comment (solveigag) [reply app.js:32]: Forgot to delete that, it's handled above

File: models/models.js [hunk 3]
```diff
@@ -78,3 +79,17 @@ exports.selectAllCommentsByArticleId = (article_id) => {
...
     });
 };
+
+exports.addCommentByArticleId = (newComment, article_id) => {
+  const { username, body } = newComment;
+
+ return checkUsernameExists(username).then((rows) => {
+    return db.query(
+      "INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *;",
+      [body, username, article_id]
```
Comment (corkesw) [inline models/models.js:89]: Comments should also have votes and created_at

**Pull request closed without merging**